app/apollo/test/cluster.fga.spec.js

```diff
@@ -0,0 +1,536 @@
+    let response;
+    try {
+      // step 1: Update the date to find the active and authorized cluster
+      response = await models.Cluster.updateMany({
+        org_id: org01._id,
+        cluster_id: testCluster1.cluster_id,
+      },
+      {$set: {
+        updated: new Moment().add(2, 'day').toDate()+1,
+      }});
+
+      // step 2: Find kubeVersionCount
+      response = await clusterApi.kubeVersionCount(fgaToken01, { orgId: org01._id });
+      const clusterCountByKubeVersion = response.data.data.clusterCountByKubeVersion;
+      expect(clusterCountByKubeVersion).to.be.an('array');
+      expect(clusterCountByKubeVersion).to.have.length(1);
+      expect(clusterCountByKubeVersion[0].id.minor).to.equal('16');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterCountByKubeVersion for fgaUser02
+  it('fgaUser02 gets zero authorized clusters when getting count of different kube versions in an org', async () => {
+    let response;
+    try {
+      response = await clusterApi.kubeVersionCount(fgaToken02, { orgId: org01._id });
+      expect(response.data.data.clusterCountByKubeVersion).to.be.an('array');
+      expect(response.data.data.clusterCountByKubeVersion).to.have.length(0);
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // deleteClusterByClusterId without authorization
+  it('fgaUser01 does NOT have authorization to delete cluster noAuthCluster by clusterID', async () => {
+    let response;
+    try {
+      response = await clusterApi.deleteClusterByClusterId(fgaToken01, {
+        orgId: org01._id,
+        clusterId: noAuthCluster.cluster_id,
+      });
+      expect(response.data.data).to.equal(null);
+      expect(response.data.errors[0].message).to.be.a('string');
+      expect(response.data.errors[0].message).to.contain('You are not allowed');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;    }
+  });
+
+  // registerCluster // deleteClusterByClusterID
+  it('fgaUser02 has authorization to create a new cluster by name, find it by name, and delete it by cluster_id', async () => {
+    let response;
+    try {
+      // step 1: register new cluster
+      response = await clusterApi.registerCluster(fgaToken02, {
+        orgId: org01._id,
+        registration: { name: 'test-cluster2-uuid' }, // Must use 'test-cluster2-uuid' for name due to fgaUser02's authorization to that value
```

Comment:
local auth was updated to support multiple strings, this should be using real name that the user is authorized to, not a 'uuid' value.

Reply:
Fixed

Reply:
Resolved